include/net/if_arp.h

```diff
@@ -0,0 +1,135 @@
+#define ARPHRD_IRDA     783             /* Linux-IrDA                   */
+
+/* ARP works differently on different FC media .. so  */
+
+#define ARPHRD_FCPP     784             /* Point to point fibrechannel  */
+#define ARPHRD_FCAL     785             /* Fibrechannel arbitrated loop */
+#define ARPHRD_FCPL     786             /* Fibrechannel public loop     */
+#define ARPHRD_FCFABRIC 787             /* Fibrechannel fabric          */
+
+/* 787->799 reserved for fibrechannel media types */
+
+#define ARPHRD_IEEE802_TR         800   /* Magic type ident for TR      */
+#define ARPHRD_IEEE80211          801   /* IEEE 802.11                  */
+#define ARPHRD_IEEE80211_PRISM    802   /* IEEE 802.11 + Prism2 header  */
+#define ARPHRD_IEEE80211_RADIOTAP 803   /* IEEE 802.11 + radiotap header */
+#define ARPHRD_IEEE802154         804   /* IEEE 802.15.4 */
+#define ARPHRD_IEEE802154_MONITOR 805   /* IEEE 802.15.4 network monitor */
+
+#define ARPHRD_PHONET      820          /* PhoNet media type            */
+#define ARPHRD_PHONET_PIPE 821          /* PhoNet pipe header           */
+#define ARPHRD_CAIF        822          /* CAIF media type              */
+#define ARPHRD_IP6GRE      823          /* GRE over IPv6                */
+#define ARPHRD_NETLINK     824          /* Netlink header               */
+#define ARPHRD_6LOWPAN     825          /* IPv6 over LoWPAN             */
+#define ARPHRD_VSOCKMON    826          /* Vsock monitor header         */
+
+#define ARPHRD_VOID        0xFFFF       /* Void type, nothing is known  */
+#define ARPHRD_NONE        0xFFFE       /* zero header length           */
+
+/* ARP protocol opcodes. */
+
+#define ARPOP_REQUEST   1               /* ARP request                  */
+#define ARPOP_REPLY     2               /* ARP reply                    */
+#define ARPOP_RREQUEST  3               /* RARP request                 */
+#define ARPOP_RREPLY    4               /* RARP reply                   */
+#define ARPOP_InREQUEST 8               /* InARP request                */
+#define ARPOP_InREPLY   9               /* InARP reply                  */
+#define ARPOP_NAK       10              /* (ATM)ARP NAK                 */
+
+/* See RFC 826 for protocol description.  ARP packets are variable
+ * in size; the arphdr structure defines the fixed-length portion.
+ * Protocol type values are the same as those for 10 Mb/s Ethernet.
+ * It is followed by the variable-sized fields ar_sha, arp_spa,
+ * arp_tha and arp_tpa in that order, according to the lengths
+ * specified.  Field names used correspond to RFC 826.
+ */
+
+struct arphdr
+{
+  unsigned short int ar_hrd;          /* Format of hardware address.  */
+  unsigned short int ar_pro;          /* Format of protocol address.  */
+  unsigned char ar_hln;               /* Length of hardware address.  */
+  unsigned char ar_pln;               /* Length of protocol address.  */
+  unsigned short int ar_op;           /* ARP opcode (command).  */
+};
```

Comment:
Why `struct arp_hdr_s` can't be used?

Reply:
glibc and musl both define arphdr, and the structure and names of the members are different from what NuttX already has. so many applications use this definition, and if we compile directly in NuttX's environment, we will get a compile error.